tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,386 @@
+    });
 
-  it('renders Extensions placeholder', () => {
-    root = renderIntoDocument(
-      <SearchFormWrapper addonType={ADDON_TYPE_EXTENSION} />).root;
-    input = root.searchQuery.input;
+    it('renders a form', () => {
+      expect(form.find('.SearchForm-form')).toHaveLength(1);
+    });
 
-    expect(input.placeholder).toEqual('Search extensions');
-    expect(input.type).toEqual('search');
-  });
+    it('renders a search input with Explore placeholder', () => {
+      expect(input).toHaveProp('placeholder', 'Search extensions and themes');
+      expect(input).toHaveProp('type', 'search');
+    });
 
-  it('renders Themes placeholder', () => {
-    root = renderIntoDocument(
-      <SearchFormWrapper addonType={ADDON_TYPE_THEME} />).root;
-    input = root.searchQuery.input;
+    it('renders Extensions placeholder', () => {
+      wrapper = mountComponent({
+        addonType: ADDON_TYPE_EXTENSION,
+      });
+      input = wrapper.find('input');
 
-    expect(input.placeholder).toEqual('Search themes');
-    expect(input.type).toEqual('search');
-  });
+      expect(input).toHaveProp('placeholder', 'Search extensions');
+      expect(input).toHaveProp('type', 'search');
+    });
 
-  it('renders the query', () => {
-    expect(input.value).toEqual('foo');
-  });
+    it('renders Themes placeholder', () => {
+      wrapper = mountComponent({
+        addonType: ADDON_TYPE_THEME,
+      });
+      input = wrapper.find('input');
 
-  it('changes the URL on submit', () => {
-    sinon.assert.notCalled(router.push);
-    input.value = 'adblock';
-    Simulate.submit(form);
-    sinon.assert.called(router.push);
-  });
+      expect(input).toHaveProp('placeholder', 'Search themes');
+      expect(input).toHaveProp('type', 'search');
+    });
 
-  it('blurs the form on submit', () => {
-    const blurSpy = sinon.stub(input, 'blur');
-    expect(!blurSpy.called).toBeTruthy();
-    input.value = 'something';
-    Simulate.submit(form);
-    sinon.assert.called(blurSpy);
-  });
+    it('renders the query', () => {
+      expect(input.prop('value')).toEqual('foo');
+    });
 
-  it('does nothing on non-Enter keydowns', () => {
-    sinon.assert.notCalled(router.push);
-    input.value = 'adblock';
-    Simulate.keyDown(input, { key: 'A', shiftKey: true });
-    sinon.assert.notCalled(router.push);
-  });
+    it('changes the URL on submit', () => {
+      sinon.assert.notCalled(router.push);
+      input.simulate('change', createFakeChangeEvent('adblock'));
+      form.simulate('submit');
+      sinon.assert.called(router.push);
+    });
 
-  it('updates the location on form submit', () => {
-    sinon.assert.notCalled(router.push);
-    input.value = 'adblock';
-    Simulate.click(root.submitButton);
-    sinon.assert.called(router.push);
-  });
+    it('blurs the form on submit', () => {
+      const blurSpy = sinon.stub(wrapper.instance().searchInput, 'blur');
+      expect(blurSpy.called).not.toBeTruthy();
```

Comment:
use `sinon.assert.notCalled()` here

Reply:
mmh actually I should have noticed that most of the `spy` rely on `sinon.assert`, not jest `expect`..